evalml/data_checks/target_leakage_data_check.py

```diff
@@ -1,69 +1,72 @@
 """Data check that checks if any of the features are highly correlated with the target by using mutual information or Pearson correlation."""
+from woodwork.config import CONFIG_DEFAULTS
+
 from evalml.data_checks import (
     DataCheck,
     DataCheckActionCode,
     DataCheckActionOption,
     DataCheckMessageCode,
     DataCheckWarning,
 )
-from evalml.utils.woodwork_utils import infer_feature_types, numeric_and_boolean_ww
+from evalml.utils.woodwork_utils import infer_feature_types
 
 
 class TargetLeakageDataCheck(DataCheck):
-    """Check if any of the features are highly correlated with the target by using mutual information or Pearson correlation.
+    """Check if any of the features are highly correlated with the target by using mutual information, Pearson correlation, and other correlation metrics.
 
-    If `method='mutual'`, this data check uses mutual information and supports all target and feature types.
-    Otherwise, if `method='pearson'`, it uses Pearson correlation and only supports binary with numeric and boolean dtypes.
-    Pearson correlation returns a value in [-1, 1], while mutual information returns a value in [0, 1].
+    If `method='mutual_info'`, this data check uses mutual information and supports all target and feature types.
+    Other correlation metrics only support binary with numeric and boolean dtypes, and return a value in [-1, 1], while mutual information returns a value in [0, 1].
```

Comment:
nit: 
"If `method='mutual_info'`, this data check uses mutual information and supports all target and feature types. 
    Other correlation metrics only support binary with numeric and boolean dtypes. This method will return a value in [-1, 1] if other correlation metrics are selected and will returns a value in [0, 1] if mutual information is selected."